### PR TITLE
Update coursier to v2.0.16

### PR DESCRIPTION
**Pull request by cheister**
Summary of release notes from https://github.com/coursier/coursier/releases
```
v2.0.16
@alexarchambault alexarchambault released this on Mar 23

Changes
New cs search command
Add new cs search command, that allows to search for install-able applications:

$ cs search meta fmt
metac
metals
metals-emacs
metap
metap-native
scalafmt
Added in #2007, thanks to @rrramiro.

Custom ClassLoader for URL protocols
In the API, add new field FileCache.classLoaders. When looking for URL handler factories for custom protocols, these class loaders are queried first.

Added in #1995, thanks to @MasseGuillaume.


v2.0.15
@alexarchambault alexarchambault released this on Mar 21

Updates
Update directories-jvm to 24


v2.0.14
@alexarchambault alexarchambault released this on Mar 16

Changes
In the CLI and the API, be fine when forcing Scala versions >= 3 (#2003, thanks to @przemek-pokrywka)
When running the JVM version of the CLI, default to forking in the launch command with Java >= 9, to work around class loader isolation issues


v2.0.13
@alexarchambault alexarchambault released this on Mar 11

Changes
Read credentials from launchers
Launchers created by cs bootstrap and cs install are now able to read and use credentials when fetching artifacts (#1991, thanks to @greg-a-atkinson)

Add aliases for Apache repositories
Use like

$ cs resolve -r apache:snapshots org.apache.spark:spark-sql_2.13:3.2.0-SNAPSHOT
Fixes
Fix some spurious NumberFormatExceptions when parsing checksums (#1972, thanks to @chimmi)


v2.0.12
@alexarchambault alexarchambault released this on Feb 15

Changes
Ignore HTTP 403 errors for optional artifacts (just like HTTP 404 errors were ignored for those - #1966, thanks to @eed3si9n). This should fix some unexpected errors in updateClassifiers once used in sbt.
New features
Add repository aliases scala-integration and scala-nightlies, both mapped to https://scala-ci.typesafe.com/artifactory/scala-integration (repository where Scala nightlies are published)


v2.0.11
@alexarchambault alexarchambault released this on Feb 12

Optimizations
Create less intermediary objects in hot paths during resolutions (#1920, thanks to @jtjeferreira)
Changes
In cs launch, if no JVM is installed, automatically download one in and use it from the coursier JVM cache, so that cs launch commands work out-of-the-box, even when no JVM is installed.


v2.0.10
@alexarchambault alexarchambault released this on Feb 12

Features
From the CLI, allow to drop default repositories by passing ! to -r, or prefixing a repository with !, like
$ cs resolve -r !central org:name:ver
$ cs resolve -r ! -r central org:name:ver
$ cs resolve -r !,central org:name:ver
Add --log-changing option logging every "changing" artifact being checked.
$ cs resolve org.scala-lang:scala-library:latest.release
Checking changing artifact file:~/.ivy2/local/org.scala-lang/scala-library/
Checking changing artifact https://repo1.maven.org/maven2/org/scala-lang/scala-library/maven-metadata.xml
org.scala-lang:scala-library:2.13.4:default
Add pure cache policy, that forbids "changing" artifacts (snapshots, version listings, …)
$ cs resolve org.scala-lang:scala-library:latest.release -m pure
Resolution error: Error downloading org.scala-lang:scala-library:latest.release
  changing artifact found: file:~/.ivy2/local/org.scala-lang/scala-library/
  changing artifact found: https://repo1.maven.org/maven2/org/scala-lang/scala-library/maven-metadata.xml
Accept application channels with a version, like
$ cs resolve ammonite --channel io.get-coursier:apps:1.0.6
Add --log-channel-version option, like
$ cs resolve ammonite --log-channel-version
Using io.get-coursier:apps-contrib:1.0.6
Using io.get-coursier:apps:1.0.6
…
Disable default app channels when a channel prefixed with ! is passed, like
$ cs resolve ammonite --channel '!io.get-coursier:apps:1.0.6'
Accept modules or dependencies as JVM channels (rather than just URLs), like
$ cs java --jvm-index io.get-coursier:jvm-index -version
…
$ cs java --jvm-index io.get-coursier:jvm-index:0.0.1 -version
…
JVM channel cs is now an alias for io.get-coursier:jvm-index (rather than an alias for a GitHub URL)
Fixes
Fix handling of aarch64 architecture in cs install command (it should now install actual aarch64 binaries, rather than x86_64 ones)


v2.0.9
@alexarchambault alexarchambault released this on Jan 19

Fixes
Ignore extra parameters like charset="UTF-8" in WWW-Authenticate headers (fixed in #1948, thanks to @cchepelov). Before these changes, credentials were not sent upon getting a parameter other than realm, so that credentials were ignored / unused in practice.


v2.0.8
@alexarchambault alexarchambault released this on Dec 23, 2020

Documentation
Many documentation additions and tweaking (#1934, #1935, #1936, #1937, #1939, all thanks to @eloots).
Updates
Update directories-jvm to 23 (#1938, thanks to @fthomas)
Miscellaneous
Internal optimization (#1921, thanks to @jtjeferreira)
Don't depend on openjdk anymore on Big Sur and Catalina in the homebrew formula (coursier/homebrew-formulas#5, coursier/homebrew-formulas#6, thanks to @kiendang and @softinio)
```

### CODE REVIEW

File: private/versions.bzl [hunk 1]
```diff
@@ -1,11 +1,11 @@
-_COURSIER_CLI_VERSION = "v2.0.7"
+_COURSIER_CLI_VERSION = "v2.0.16"
 
 COURSIER_CLI_HTTP_FILE_NAME = ("coursier_cli_" + _COURSIER_CLI_VERSION).replace(".", "_").replace("-", "_")
 COURSIER_CLI_GITHUB_ASSET_URL = "https://github.com/coursier/coursier/releases/download/{COURSIER_CLI_VERSION}/coursier.jar".format(COURSIER_CLI_VERSION = _COURSIER_CLI_VERSION)
 
 # Run 'bazel run //:mirror_coursier' to upload a copy of the jar to the Bazel mirror.
```
Comment (cheister) [inline private/versions.bzl:6]: @jin I don't have permission to run this, can you upload this version to the bazel mirror?

Comment (jin) [reply private/versions.bzl:6]: I've lost access too:

```
$ bazelisk run //:mirror_coursier
DEBUG: /usr/local/google/home/jingwen/.cache/bazel/_bazel_jingwen/8484bc4fff18ee4a905b69a9ddb0e143/external/bazel_toolchains/rules/rbe_repo/version_check.bzl:68:14:
Current running Bazel is ahead of bazel-toolchains repo. Please update your pin to bazel-toolchains repo in your WORKSPACE file.
DEBUG: /usr/local/google/home/jingwen/.cache/bazel/_bazel_jingwen/8484bc4fff18ee4a905b69a9ddb0e143/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: buildkite_config not using checked in configs; Bazel version 4.1.0 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"], "0.29.0": ["9.0.0"], "0.29.1": ["9.0.0", "10.0.0"], "1.0.0": ["9.0.0", "10.0.0"], "1.0.1": ["10.0.0"], "1.1.0": ["10.0.0"], "1.2.0": ["10.0.0"], "1.2.1": ["10.0.0"], "2.0.0": ["10.0.0"], "2.1.0": ["10.0.0"], "2.1.1": ["10.0.0", "11.0.0"], "2.2.0": ["11.0.0"], "3.0.0": ["11.0.0"], "3.1.0": ["11.0.0"], "3.2.0": ["11.0.0"], "3.3.0": ["11.0.0"], "3.3.1": ["11.0.0"], "3.4.1": ["11.0.0"], "3.5.0": ["11.0.0"]}
INFO: Analyzed target //:mirror_coursier (1 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //:mirror_coursier up-to-date:
  bazel-bin/mirror_coursier
INFO: Elapsed time: 0.198s, Critical Path: 0.01s
INFO: 2 processes: 2 internal.
INFO: Build completed successfully, 2 total actions
INFO: Build completed successfully, 2 total actions
Copying file://external/coursier_cli_v2_0_16/file/downloaded [Content-Type=application/octet-stream]...
ResumableUploadAbortException: 403 jingwen@google.com does not have storage.objects.create access to the Google Cloud Storage object.
```

@philwo could you please re-grant access to me to `gs://bazel-mirror/coursier_cli`?

Comment (jin) [reply private/versions.bzl:6]: (internal bug: b/196755452)

Comment (philwo) [reply private/versions.bzl:6]: Fixed!

Comment (jin) [reply private/versions.bzl:6]: Success:

```
Copying file://external/coursier_cli_v2_0_16/file/downloaded [Content-Type=application/octet-stream]...
\ [1 files][ 34.2 MiB/ 34.2 MiB]
Operation completed over 1 objects/34.2 MiB.
```

Thanks @philwo!

**Review by jin — APPROVED** [review]

**Pull request merged**